app/controllers/clearance/passwords_controller.rb

```diff
@@ -88,18 +87,19 @@ def flash_failure_after_update
       default: t('flashes.failure_after_update'))
   end
 
-  def forbid_missing_token
-    if params[:token].to_s.blank?
+  def check_for_forbidden_access
+    if missing_token? || user_does_not_exist?
```

Comment:
What do you think about something like:

```ruby
unless find_user_by_id_and_confirmation_token
  #...
end
```

And then moving the empty param check into `find_user_by_id_and_confirmation_token` itself? Would save us a couple of private methods and is more in line with what we want, I think - basically you shouldn't ever find a user here without them having a cofnirmation token set.